lxc/utils_properties.go

```diff
@@ -227,3 +227,60 @@ func unpackKVToWritable(writable any, keys map[string]string) error {
 
 	return nil
 }
+
+// getEditableYAMLFields uses reflection to extract YAML field names from a struct.
+func getEditableYAMLFields(v any) []string {
+	var fields []string
+	typ := reflect.TypeOf(v)
+
+	// Handle pointer types
+	if typ.Kind() == reflect.Ptr {
+		typ = typ.Elem()
+	}
+
+	// Only process struct types
+	if typ.Kind() != reflect.Struct {
+		return fields
+	}
+
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		yamlTag := field.Tag.Get("yaml")
+
+		// Skip fields without yaml tags or with "-"
+		if yamlTag == "" || yamlTag == "-" {
+			continue
+		}
+
+		// Extract the field name from the yaml tag (handle "name,omitempty" format)
+		parts := strings.Split(yamlTag, ",")
+		if len(parts) > 0 && parts[0] != "" {
+			fields = append(fields, parts[0])
+		}
+	}
+
+	return fields
+}
+
+// formatFieldList formats a slice of field names as a quoted, comma-separated list.
+func formatFieldList(fields []string) string {
+	if len(fields) == 0 {
+		return ""
+	}
+
+	quoted := make([]string, len(fields))
```

Comment:
please avoid prefilling the slice with zero entries like this, and instead use `make([]string,0,len(fields))` to pre-create with right capacity but without zero entries.

Then use `append` in the `for` loop.